src/agora/txrx/txrx_usrp.cpp

```diff
@@ -0,0 +1,360 @@
+        cfg->sampsPerSymbol * 14, 0);
+    std::vector<std::complex<int16_t>> samp_buffer1(
+        cfg->sampsPerSymbol * 14, 0);
+    std::vector<void*> samp_buffer(2);
+    samp_buffer[0] = samp_buffer0.data();
+    if (true)
+        samp_buffer[1] = samp_buffer1.data();
+
+    // long long rxTimeBs(0);
+    // long long txTimeBs(0);
+    rxTimeBs = 0;
+    txTimeBs = 0;
+
+    std::cout << "Sync BS host and FGPA timestamp..." << std::endl;
+    radioconfig_->radioRx(0, samp_buffer.data(), rxTimeBs);
+    // Schedule the first beacon in the future
+    txTimeBs = rxTimeBs + cfg->sampsPerSymbol * cfg->symbol_num_perframe * 40;
+    radioconfig_->radioTx(0, beaconbuff.data(), 2, txTimeBs);
+    long long bsInitRxOffset = txTimeBs - rxTimeBs;
+    for (int it = 0; it < std::floor(bsInitRxOffset / cfg->sampsPerSymbol);
+         it++) {
+        radioconfig_->radioRx(0, samp_buffer.data(), rxTimeBs);
+    }
+
+    std::cout << std::endl;
+    std::cout << "Init BS sync done..." << std::endl;
+    std::cout << "Start BS main recv loop..." << std::endl;
+
+    int global_frame_id = 0;
+    int global_symbol_id = 0;
+
+    int prev_frame_id = -1;
+    int radio_id = radio_lo;
+    while (cfg->running) {
+
+        // transmit data
+        // if (-1 != dequeue_send_usrp(tid))
+        //   continue;
+        // receive data
+        // struct Packet* pkt = recv_enqueue_usrp(tid, radio_id, rx_offset);
+        struct Packet* pkt = recv_enqueue_usrp(
+            tid, radio_id, rx_offset, global_frame_id, global_symbol_id);
+
+        // Schedule beacon in the future
+        if (global_symbol_id == 0) {
+            txTimeBs = rxTimeBs
+                + cfg->sampsPerSymbol * cfg->symbol_num_perframe * 20;
+            int tx_ret
+                = radioconfig_->radioTx(0, beaconbuff.data(), 2, txTimeBs);
+            if (tx_ret != (int)cfg->sampsPerSymbol)
+                std::cerr << "BAD Transmit(" << tx_ret << "/"
+                          << cfg->sampsPerSymbol << ") at Time " << txTimeBs
+                          << ", frame count " << global_frame_id << std::endl;
+        }
+
+        if (++radio_id == radio_hi)
+            radio_id = radio_lo;
+
+        // Update global frame_id and symbol_id
+        global_symbol_id++;
+        if (global_symbol_id == (int)cfg->symbol_num_perframe) {
+            global_symbol_id = 0;
+            global_frame_id++;
+        }
+
+        if (pkt == NULL)
+            continue;
+        rx_offset = (rx_offset + cfg->nChannels) % packet_num_in_buffer_;
+
+        if (kIsWorkerTimingEnabled) {
+            int frame_id = pkt->frame_id;
+            // int symbol_id = pkt->symbol_id;
+            if (frame_id > prev_frame_id) {
+                rx_frame_start[frame_id % kNumStatsFrames] = rdtsc();
+                prev_frame_id = frame_id;
+            }
+        }
+    }
+    return 0;
+}
+
+struct Packet* PacketTXRX::recv_enqueue_usrp(
+    int tid, int radio_id, int rx_offset)
```

Comment:
Do you need to keep this function? It seems only this one is used.
```
struct Packet* recv_enqueue_usrp(
     int tid, int radio_id, int rx_offset, int frame_id, int symbol_id);
```

Reply:
removed!